sdk/identity/identity/src/credentials/azureCliCredential.ts

```diff
@@ -34,14 +34,14 @@ export class AzureCliCredential implements TokenCredential {
    * Gets the access token from Azure CLI
    * @param resource The resource to use when getting the token
    */
-  protected async getAzureCliAccessToken(resource: string) {
+  protected async getAzureCliAccessToken(resource: string): Promise<unknown> {
```

Comment:
Any reason to choose `unknown` here instead of say inlining the type of the object containing the stdout, stderr and error?

Reply:
Currently in `identity.api.md` the return type of this method is `Promise<unknown>` and I was not sure if we should change the public API. Initially, I thought of setting it to `Promise<any>`

Reply:
I believe changing from unknown to a strongly typed value especially in the output of a method should not be a breaking change. Doing so from `any` would be breaking though. 

cc @xirzec to confirm

Reply:
Please review this [commit](https://github.com/Azure/azure-sdk-for-js/pull/11453/commits/1c576275c03ea07a2a0db66b43e276482d29083a)

Reply:
Marking this conversation as unresolved to ensure we have input from @xirzec here

Reply:
In-lining the type is better than `unknown` and `any`, though a small type alias or interface would be even better. I'm good with the linked commit though.

In this particular case, I wonder how much we have to worry about breaking changes for protected members. It seems like the only reason this method is protected is out of a misguided strategy for testing. We should be using something like sinon to mock out `getAzureCliAccessToken` instead of inheritance. 

If @jonathandturner is willing, perhaps we could make this private unless we really do anticipate developers to replace this implementation in their live apps?